matter_server/server/device_controller.py

```diff
@@ -632,3 +629,32 @@ def _parse_attributes_from_read_result(
                     )
                     result[attribute_path] = attr_value
         return result
+
+    async def _resolve_node(
+        self, node_id: int, retries: int = 3, allow_pase: bool = False
+    ) -> None:
+        """Resolve a Node on the network."""
+        if self.chip_controller is None:
+            raise RuntimeError("Device Controller not initialized.")
+        try:
+            if allow_pase:
+                # last attempt allows PASE connection (last resort)
+                LOGGER.debug(
+                    "Attempting to resolve node %s (with PASE connection)", node_id
+                )
+                await self._call_sdk(
+                    self.chip_controller.GetConnectedDeviceSync,
+                    nodeid=node_id,
+                    allowPASE=True,
+                )
+                return
+            LOGGER.debug("Resolving node %s", node_id)
+            await self._call_sdk(self.chip_controller.ResolveNode, nodeid=node_id)
+        except (ChipStackError, TimeoutError) as err:
+            if not retries:
+                # when we're out of retries, raise NodeNotResolving
+                raise NodeNotResolving(f"Unable to resolve Node {node_id}") from err
+            await self._resolve_node(
+                node_id=node_id, retries=retries - 1, allow_pase=retries == 0
```

Comment:
```suggestion
                node_id=node_id, retries=retries - 1, allow_pase=retries - 1== 0
```

Reply:
good catch, we want 3 attempts total. 2 of them are without PASE and the last one is with